src/Core/Grid/Definition/Factory/CartGridDefinitionFactory.php

```diff
@@ -0,0 +1,323 @@
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\CartBadgeColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\CartBulkActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\DateTimeColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\DataColumn;
+use PrestaShop\PrestaShop\Core\Grid\Filter\Filter;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollection;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShopBundle\Form\Admin\Type\DatalistType;
+use PrestaShopBundle\Form\Admin\Type\DateRangeType;
+use PrestaShopBundle\Form\Admin\Type\SearchAndResetType;
+use PrestaShopBundle\Form\Admin\Type\YesAndNoChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\NumberType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+/**
+ * Creates grid definition for Carts grid
+ */
+final class CartGridDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    const GRID_ID = 'cart';
+
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @var string
+     */
+    private $contextDateFormat;
+
+    /**
+     * @var AccessibilityCheckerInterface
+     */
+    private $deleteCartAccessibilityChecker;
+
+    /**
+     * @param HookDispatcherInterface $dispatcher
+     * @param ConfigurationInterface $configuration
+     * @param AccessibilityCheckerInterface $deleteCartAccessibilityChecker
+     * @param string $contextDateFormat
+     */
+    public function __construct(
+        HookDispatcherInterface $dispatcher,
+        ConfigurationInterface $configuration,
+        AccessibilityCheckerInterface $deleteCartAccessibilityChecker,
+        $contextDateFormat
+    ) {
+        parent::__construct($dispatcher);
+
+        $this->configuration = $configuration;
+        $this->contextDateFormat = $contextDateFormat;
+        $this->deleteCartAccessibilityChecker = $deleteCartAccessibilityChecker;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return $this->trans('Shopping Carts', [], 'Admin.Navigation.Menu');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        $columns = (new ColumnCollection())
+            ->add((new CartBulkActionColumn('cart_bulk_action'))
+                ->setOptions([
+                    'bulk_field' => 'id_cart',
+                ])
+            )
+            ->add((new DataColumn('id_cart'))
+                ->setName($this->trans('ID', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'id_cart',
+                ])
+            )
+            ->add((new DataColumn('status'))
```

Comment:
I think the special template (therefor column) was meant for this field (hence the presence of `record.status` in the template)
The `cart_total` field has a rather conventional behavior But status is a mess because it can contain either an order_id or a string status, that's why the template was messy as well I guess what we really need is more a `CartStatusColumn` than a `CartBadgeColumn`